packages/react-reconciler/src/ReactProfilerTimer.js

```diff
@@ -55,6 +56,7 @@ if (__DEV__) {
   fiberStack = [];
 }
 
+let rootStartTimes: Map<FiberRoot | null, number> = new Map();
```

Comment:
I think I can replace this Map with a `batchStartTime` number field. The only case we really need to handle is processing a batch commit while async work is yielded. Other root changes will unwind the yielded work, which w already handled.

This was just a hacky first attempt to make tests pass. 😄